python/lsst/faro/base/MatchedCatalogBase.py

```diff
@@ -264,9 +278,25 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
                 calib_find = (visitSkyWcsList == visit) & (
                     detectorSkyWcsList == detector
                 )
-                row = flatSkyWcsList[calib_find]
-                externalSkyWcs = row[0].getWcs()
-                inputs["astromCalibs"][i] = externalSkyWcs
+                if np.sum(calib_find) < 1:
+                    self.log.warning("Detector id %s not found in externalSkyWcsCatalog "
+                                     "for visit %s and will not be used.",
+                                     detector, visit)
+                    inputs["astromCalibs"][i] = None
+                    remove_indices.append(i)
+                else:
+                    row = flatSkyWcsList[calib_find]
+                    externalSkyWcs = row[0].getWcs()
+                    inputs["astromCalibs"][i] = externalSkyWcs
+
+        # Remove datasets that didn't have matching external calibs
+        remove_indices = np.unique(np.array(remove_indices))
+        if len(remove_indices) > 0:
+            for ind in sorted(remove_indices, reverse=True):
```

Comment:
It looks like this step is to ensure that the datasets are removed starting from the end of the array so that the index positions of elements at the start of the array are not affected.

Reply:
Yeah, if you don't start from the end and work backwards, then the indices get messed up.